appinventor/appengine/src/com/google/appinventor/client/actions/SendToGalleryAction.java

```diff
@@ -25,40 +25,55 @@ public class SendToGalleryAction implements Command {
 
   @Override
   public void execute() {
+    if (lockPublishButton) {
+      return;                   // De-bounce the publish button
+    }
+    lockPublishButton = true;
```

Comment:
It's probably a bit pedantic, but if an exception gets thrown anywhere in the code below it's possible that the `lockPublishButton` will never be reset. For example, what if `getCurrentYoungAndroidProjectRootNode()` returns null, causing the `hasExtensions` call to trigger a NPE?

Since `lockPublishButton` is set to `true` in `sendToGallery`, this is redundant and probably should be removed.